Assets/Scripts/ThrowController.cs

```diff
@@ -0,0 +1,216 @@
+    [Range(1.0f, 10.0f)] public float sensitivityController = 2.0f;
+    public Transform throwPosition;
+    public float throwableDestroyTime = 5f;
+    public float throwMultiplier = 0.08f;
+    public float minThrowVelocity = 10f;
+    public float maxThrowVelocity = 20f;
+
+    private LaunchArcRenderer launchArcRenderer;
+    private List<GameObject> currentThrowables;
+    private Plane mouseHitPlane;
+    private Camera mainCamera;
+    private bool mainCameraActive = false;
+    private bool isThrowReset = true;
+
+    public delegate void OnThrowEventHandler(Interactable source);
+    public event OnThrowEventHandler OnThrow;
+
+    public delegate void OnPickupEventHandler(GameObject source);
+    public event OnPickupEventHandler OnPickup;
+    
+    public delegate void OnThrowableResetEventHandler();
+    public event OnThrowableResetEventHandler OnThrowableReset;
+
+    public delegate void OnInteractEventHandler(DialogueInteractable source);
+    public event OnInteractEventHandler OnInteractBegin;
+
+    private void Start()
+    {
+        launchArcRenderer = GetComponentInChildren<LaunchArcRenderer>();
+        launchArcRenderer.gameObject.SetActive(false);  // initially hide arc
+        currentThrowables = new List<GameObject>();
+        
+        mouseHitPlane = new Plane(Vector3.up, Vector3.zero);
+        mainCamera = Camera.main;
+
+        UIManager.Instance.staminaBar.OnChange += UpdateThrowVelocity;
+    }
+
+    private void Update()
+    {
+        HandleThrowInput();
+    }
+
+    private void UpdateThrowVelocity(float fillAmount)
+    {
+        launchArcRenderer.velocity = Mathf.Lerp(minThrowVelocity, maxThrowVelocity, fillAmount / StaminaBar.FILL_MAX);
+    }
+
+    public Vector3 GetMousePosition()
+    {
+        if (mainCameraActive)
+        {
+            Vector3 position = transform.position;
+            
+            // Handle controller input
+            if (GameManager.Instance.GetPlayerController().controllerConnected && 
+                !Mathf.Approximately(Input.GetAxis("Horizontal Right Stick") + Input.GetAxis("Vertical Right Stick"),
+                    0f))
+            {
+                float horizontal = 10 * sensitivityController * Input.GetAxis("Vertical Right Stick");
+                float vertical = 10 * sensitivityController * Input.GetAxis("Horizontal Right Stick");
+                
+                return new Vector3(position.x + horizontal, position.y, position.z + vertical);
+            }
+            
+            // Handle mouse + keyboard input
+            Ray ray = mainCamera.ScreenPointToRay(Input.mousePosition);
+                    
+            if(mouseHitPlane.Raycast(ray, out float enter))
+            {
+                Vector3 hitPoint = ray.GetPoint(enter);
+                
+                float horizontal = sensitivityMouse * (hitPoint.x - position.x);
+                float vertical = sensitivityMouse * (hitPoint.z - position.z);
```

Comment:
Maybe the issue with different sensitivities is that we have to clamp this `(hitPoint.x - position.x)` value, since it can be pretty far too. Between 0 and 1, since Input.GetAxis is always between 0 and 1.